viite-backend/viite-main/src/main/scala/fi/liikennevirasto/viite/process/strategy/TerminatedTwoTrackSectionSynchronizer.scala

```diff
@@ -0,0 +1,473 @@
+
+    val continuousAfterTerminatedLeft   = projectLinks.find(pl => pl.track == Track.LeftSide && pl.originalAddrMRange.continuesFrom(lastTerminatedOnLeftSideSection.originalAddrMRange))
+    val continuousAfterTerminatedRight  = projectLinks.find(pl => pl.track == Track.RightSide && pl.originalAddrMRange.continuesFrom(lastTerminatedOnRightSideSection.originalAddrMRange))
+
+    // Adjust terminated
+    val processedLinks: Seq[ProjectLink] = {
+      if (Math.abs(lastTerminatedOnLeftSideSection.addrMRange.end - lastTerminatedOnRightSideSection.addrMRange.end) <= maxDiffForTrackEnds) {
+        // Adjust the last terminated links on each track to match
+        val (adjustedTerminatedLeft, adjustedTerminatedRight, averagedAddrMRange) = adjustTerminatedToMatch(lastTerminatedOnLeftSideSection, lastTerminatedOnRightSideSection)
+
+        if (continuousAfterTerminatedLeft.isDefined && continuousAfterTerminatedRight.isDefined) {
+          // Adjust both links and update the project links list with the adjusted terminated links and the adjusted continuous-after-termination links
+          val adjustedLeftAddrMRange = AddrMRange(averagedAddrMRange.end, continuousAfterTerminatedLeft.get.addrMRange.end)
+          val adjLeftContinuousAfterTerminated = continuousAfterTerminatedLeft.get.copy(addrMRange = adjustedLeftAddrMRange,originalAddrMRange = adjustedLeftAddrMRange)
+
+          val adjustedRightAddrMRange = AddrMRange(averagedAddrMRange.end, continuousAfterTerminatedRight.get.addrMRange.end)
+          val adjRightContinuousAfterTerminated = continuousAfterTerminatedRight.get.copy(addrMRange = adjustedRightAddrMRange, originalAddrMRange = adjustedRightAddrMRange)
+
+          updateProjectLinksList(Seq(adjustedTerminatedLeft, adjustedTerminatedRight, adjLeftContinuousAfterTerminated, adjRightContinuousAfterTerminated), projectLinks)
+        } else {
+          // Just update the terminated links
+          updateProjectLinksList(Seq(adjustedTerminatedLeft, adjustedTerminatedRight), projectLinks)
+        }
+
+      } else {
+        projectLinks
+      }
+    }
+    processedLinks
+  }
+
+  def roadPartMiddleTerminated(projectLinks: Seq[ProjectLink]): Seq[ProjectLink] = {
+
+    val terminatedLinks = projectLinks.filter(_.status == RoadAddressChangeType.Termination)
+
+    val (leftTerminatedSections, rightTerminatedSections) = terminatedLinksToToContinuousTwoTrackSections(terminatedLinks)
+
+    val processedLinks = if (leftTerminatedSections.nonEmpty && rightTerminatedSections.nonEmpty) { // Check if road part start is terminated on both tracks
+      handleTwoTrackMiddleTermination(leftTerminatedSections, rightTerminatedSections, projectLinks)
+    } else
+      projectLinks
+
+    processedLinks
+  }
+
+  /**
+   * Adjusts two track terminated sections to match if the termination creates a minor discontinuity.
+   * The minor discontinuity links will also be adjusted to match the start of the terminated section.
```

Comment:
"The minor discontinuity links" ->"The minor discontinuity links (links just before the terminated sections)"